### ISSUE TITLE
nimble support

### ISSUE THREAD

**Issue by FlorianLudwig** [dbSuS/libpax#22]
Hi @cyberman54  / @oliverbrandmueller

I would like to suggest to add back a build-time switch to toggle between nimble.

The use-case is we would like to use the library in a context where we also use BLE via nimble. With nimble-based libpax it is possible to use nimble while also having libpax running.

Is that something you would accept in libpax?

**Comment by oliverbrandmueller** [comment]
Though this brings in a little more complexity, I'd accept this approach. Also for future use cases, which might imply to use BLE for other purposes in the same device, this opens up an easy and fast way to imlement this.

- the build switch **MUST** completely disable using/linking the nimble library, if unused (regading: code size, security issues)
- If possibe, the change **SHOULD** be backwards compatible, so we don't need a major version bump. However, I would accept a version bump, if too many workarounds would be needed.

**Comment by FlorianLudwig** [comment]
Thanks for the quick replies :)

@oliverbrandmueller both of your requirements should be possible to be met.

@cyberman54 
> We should keep this structure maintained, because it has a good fit for embedded systems, i.e. it allows to implement a bluetooth host stack on a different device than that which libpax is running on.

As far as I know that is currently not possible. The host stack operates under the assumption it is the only one issuing commands to the hardware. So only one host stack can be active at any time. With libpax talking directly to the hardware via VHCI it basically acts like a host stack, making it error prone to run another host stack in parallel.

**Comment by cyberman54** [comment]
That's true. There is no kind of thread safe logic. And the VHCI is working asynchronous, what makes things more complex. Thus, it won't make sense to simply top up a Bluetooth stack. So we need an integration.
